config/main.py

```diff
@@ -1152,25 +1153,69 @@ def validate_gre_type(ctx, _, value):
         return gre_type_value
     except ValueError:
         raise click.UsageError("{} is not a valid GRE type".format(value))
-    
+
 # Function to apply patch for a single ASIC.
 def apply_patch_for_scope(scope_changes, results, config_format, verbose, dry_run, ignore_non_yang_tables, ignore_path):
     scope, changes = scope_changes
     # Replace localhost to DEFAULT_NAMESPACE which is db definition of Host
     if scope.lower() == "localhost" or scope == "":
         scope = multi_asic.DEFAULT_NAMESPACE
-        
+
     scope_for_log = scope if scope else "localhost"
     try:
         # Call apply_patch with the ASIC-specific changes and predefined parameters
-        GenericUpdater(namespace=scope).apply_patch(jsonpatch.JsonPatch(changes), config_format, verbose, dry_run, ignore_non_yang_tables, ignore_path)
+        GenericUpdater(scope=scope).apply_patch(jsonpatch.JsonPatch(changes),
+                                                config_format,
+                                                verbose,
+                                                dry_run,
+                                                ignore_non_yang_tables,
+                                                ignore_path)
         results[scope_for_log] = {"success": True, "message": "Success"}
         log.log_notice(f"'apply-patch' executed successfully for {scope_for_log} by {changes}")
     except Exception as e:
         results[scope_for_log] = {"success": False, "message": str(e)}
         log.log_error(f"'apply-patch' executed failed for {scope_for_log} by {changes} due to {str(e)}")
 
 
+def validate_patch(patches):
+    command = "show runningconfiguration all"
+    proc = subprocess.Popen(command, text=True, stdout=subprocess.PIPE)
+    all_running_config, returncode = proc.communicate()
+    if returncode:
+        log.log_notice(f"Fetch all runningconfiguration failed as output:{all_running_config}")
+        return False
+
+    try:
+        # Duplicate check
+        resource_usage = {}
+        for patch in patches:
+            resource_path = patch['path']
+            if resource_path in resource_usage:
+                log.log_notice(f"Conflict detected: multiple patches modify {resource_path}")
+                return False
+            resource_usage[resource_path] = patch
+
+        # Structure validation and simulate apply patch.
```

Comment:
>simulate apply patch. [](http://example.com/codeflow?start=35&length=21)

Is it possible to reuse existing code, or form a new library function?

Reply:
exist code is for single asic scope, not the entire box level. Also, the exist code also just one line which is applying jsonpatch for current asic config db json.